### DISCUSSION TITLE
Requirement for Clock accuracy for CSL feature

### DISCUSSION THREAD

**Discussion by AhmedEsmailm** [openthread/openthread#9514] [Q&A] [answered]
Hello OT Community,

I understand that there is no specific requirement for clock accuracy before the implementation the CSL feature. I am wondering about the requirements for clock accuracy for CLS feature. Is there a specific requirement?
Also, I see the max value for clock uncertainty is 255 (max of uint8_t range "The return value of otPlatRadioGetCslAccuracy API). I believe that CSL should work with clock with larger PPM  clock error. for example 500PPM like the case of BLE. How  can  this be represented in  uint8_t range?
Thanks,
Ahmed

**Comment by Irving-cl** [comment, marked as answer]
Hi @AhmedEsmailm

I'm not sure what's the worst accuracy that still keeps CSL work. CSL itself takes Accuracy into consideration: 
> The CSL Transmitter provides clock accuracy to the end device via the MLE CSL Accuracy TLV. A
CSL Receiver MAY take this and its own local clock drift into consideration when determining how
often to synchronize.

That means, the worse accuracy the Transmitter or Receiver has, the larger extra sample window the receiver will have. This is implemented in OT's implementation.

> I believe that CSL should work with clock with larger PPM clock error. for example 500PPM like the case of BLE. How can this be represented in uint8_t range?

Currently in OT, accuracy larger than 255 cannot be supported. When designing this, we don't expect accuracy larger than that. From our experience, the accuracy should be small to make this work. Otherwise it means we have to increase the receiver's sample window, which is not a good thing.
